### PR TITLE
Daml assitant support for Http metering endpoint [DPP-858]

### PR DESCRIPTION
**Pull request by simonmaxen-da**
changelog_begin
changelog_end

Note that this PR is dependent upon https://github.com/digital-asset/daml/pull/13147

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Nice work thank you!

File: language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs [hunk 1]
```diff
@@ -29,6 +30,22 @@ import GHC.Word (Word32)
 import Data.Time.Calendar (Day(..))
 import Data.Time.Clock (secondsToDiffTime, UTCTime(..))
 
+data MeteringRequestByDay = MeteringRequestByDay {
+  from :: Day
+, to :: Maybe Day
+, application :: Maybe ApplicationId
+} deriving (Show, Eq)
+
+instance ToJSON MeteringRequestByDay where
```
Comment (cocreature) [inline language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs:39]: I think you can kill the `ToJSON` instance of `MeteringRequest` in that case.

Comment (cocreature) [reply language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs:39]: actually do we need this type at all now?

**Review by cocreature — APPROVED** [review]
changes look

File: language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs (hunk 1 above)
Comment (simonmaxen-da) [reply language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs:39]: `MeteringRequest` is still used in the `MeteringReport` and is both received from the Http service _and_ used to serialize the output json for both underlying transports.

The `MeteringRequestByDay` is only sent to the Http service

Comment (simonmaxen-da) [reply language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs:39]: So we need to keep both the `ToJSON` instance and both request variants.

**Pull request merged**